### PR TITLE
Fix non-value type variables marked as immutable candidate

### LINKED ISSUE #75: Non-value type variable should not be marked as immutable candidate
Currently, a non-value type variable is marked as an immutable candidate, but immutability is not available for such variables and they must be defined as constants instead.

[https://docs.soliditylang.org/en/v0.8.13/contracts.html?highlight=immutable#constant-and-immutable-state-variables](https://docs.soliditylang.org/en/v0.8.13/contracts.html?highlight=immutable#constant-and-immutable-state-variables)

### PR DESCRIPTION
**Pull request by dijkstra-dev**
Fix #75

### CODE REVIEW

File: src/analyzer/optimizations/immutable_variables.rs [hunk 1]
```diff
@@ -278,6 +288,35 @@ pub fn get_storage_variables_assigned_in_constructor(
     potential_immutable_variables
 }
 
+fn is_a_non_value_type(assigned_value: Box<pt::Expression>) -> bool {
+    // println!("{:?}", assigned_value);
```
Comment (0xKitsune) [inline src/analyzer/optimizations/immutable_variables.rs:292]: Please remove the commented out `println!`.

**Comment by 0xKitsune** [comment]
Hey thanks for the PR and nice catch! 

Only one small change, can you remove the commented out print statement on line 292? Other than that we should be good to merge this into dev. Ill be merging dev into main before the weekend ends.

File: src/analyzer/optimizations/immutable_variables.rs (hunk 1 above)
Comment (dijkstra-dev) [reply src/analyzer/optimizations/immutable_variables.rs]: Fixed

**Comment by dijkstra-dev** [comment]
Ups!Sure! already fixed. Thanks!

**Pull request merged**